enterprise/cloud.oracle/src/org/netbeans/modules/cloud/oracle/assets/RunInClusterAction.java

```diff
@@ -0,0 +1,281 @@
+        ProgressHandle h = ProgressHandle.createHandle("aaa");
+        try {
+            h.start();
+            
+            Project project = Steps.getDefault()
+                    .executeMultistep(new ProjectStep(), Lookup.EMPTY)
+                    .thenApply(values -> {
+                        return values.getValueForStep(ProjectStep.class);
+                    })
+                    .get();
+            ProjectInformation pi = ProjectUtils.getInformation(project);
+            String projectName = pi.getDisplayName();
+
+            ClusterItem cluster = CloudAssets.getDefault().getItem(ClusterItem.class);
+            if (cluster.getConfig() == null) {
+                cluster.update();
+            }
+
+            runWithClient(cluster, client -> {
+                Deployment existingDeployment = null;
+                DeploymentList dList = client.apps().deployments().list();
+                for (Deployment deployment : dList.getItems()) {
+                    if (projectName.equals(deployment.getMetadata().getName())) {
+                        existingDeployment = deployment;
+                    }
+                }
+                if (existingDeployment != null) {
+                    client.apps()
+                            .deployments()
+                            .inNamespace(DEFAULT)
+                            .withName(projectName)
+                            .edit(d -> new DeploymentBuilder(d)
+                            .editSpec()
+                            .editTemplate()
+                            .editSpec()
+                            .editFirstContainer()
+                            .withImage(context.getUrl()) // New image version
+                            .endContainer()
+                            .endSpec()
+                            .endTemplate()
+                            .endSpec()
+                            .build());
+
+                    client.apps()
+                            .deployments()
+                            .inNamespace(DEFAULT)
+                            .withName(projectName)
+                            .edit(d -> new DeploymentBuilder(d)
+                            .editSpec()
+                            .editTemplate()
+                            .editMetadata()
+                            .addToAnnotations("kubectl.kubernetes.io/restartedAt", Instant.now().toString()) //NOI18N
+                            .endMetadata()
+                            .endTemplate()
+                            .endSpec()
+                            .build());
+
+                } else {
+                    Deployment newDeployment = new DeploymentBuilder()
+                            .withNewMetadata()
+                            .withName(projectName)
+                            .addToLabels(APP, projectName)
+                            .endMetadata()
+                            .withNewSpec()
+                            .withReplicas(3)
```

Comment:
Nitpick: make configuration constants from these (replicas, port, ...)
I wonder if this could be somehow parametrized, for the case the user/LSP client/API client would need a different shape of the deployment.

Reply:
Parametrisation will be addressed i subsequent PRs